src/classes/STG_InstallScript.cls

```diff
@@ -51,17 +55,27 @@ global without sharing class STG_InstallScript implements InstallHandler {
             } else if(context.isPush()) {
                 insertTdtmDefaults(null);
             }
+
             UTIL_MasterSchedulableHelper.abortOldScheduledJobs();
+
+            rethrowException = true;
+
+            if (context.previousVersion() == null || context.previousVersion().compareTo(new Version(3, 79)) < 0) {
+                // update record type settings ONLY if upgrading from a version prior to 3.78
+                UTIL_RecordTypeSettingsUpdate.getInstance().updateSettings();
+            }
+
         } catch(Exception e) {
+            Database.rollback(sp);
+
+            if (rethrowException) {
```

Comment:
it's not clear to me that the original intent was if the updateSettings() hit an error, that we'd want the install script to throw the exception.  I just wonder if it was accidentally added after the try/catch block, rather than being intentional.  Can you please talk to @dospromptman, who added this settings work, about a year ago, to see if he recalls whether the intent was for this to stop the install, rather than just log the problem like other install errors?  thanks.

Reply:
Yes, it was intentional.  Here is a link to the comment and pull request that talks about moving it outside the try/catch: https://github.com/SalesforceFoundation/Cumulus/pull/2184#issuecomment-236970992

I can't find the original conversation we had around this-- it probably happened in HipChat.  But the reasoning was that if this conversion process from Record Type Name to Record Type Id failed, then the user could possibly lose data and regardless the behavior of NPSP in their org would not be correct (due to the code expecting the settings to be in the new format).

So, if the setting conversion process failed for any reason then we would want the push upgrade to fail so we didn't put the user's org in a state where NPSP would behave incorrectly.  The way to fail the install was to allow the exception to remain unhandled.

As I recall, there were some instances where it did actually did cause the push upgrade to fail in customer orgs.  I don't remember the exact reasons, but I think it was something about the length limit of the field and the converted settings not fitting?  I think the failed install helped identify the issue and the customer fixed it and was able to get upgraded without losing data.

Reply:
Thanks for the info @dospromptman !  glad to hear it was intentional.  @RadSFDeveloper can you just add a quick comment in the code that we explicitly intend for a failure in the settings update to abort the upgrade.  thanks.

Reply:
The comment about re-throwing exception to stop the package installation script has been added. Please let me know if needs to be changed more.